include/coap/coap_dtls.h

```diff
@@ -33,6 +33,63 @@ void coap_dtls_set_log_level(int level);
 /** Returns the current log level. */
 int coap_dtls_get_log_level(void);
 
+/** The structure used for defining the PKI setup data to be used */
+typedef struct coap_dtls_pki_t {
+  /* Alternative 1: Name of file on disk */
+  const char *ca_file;
+  const char *public_cert;
+  const char *private_key;
+  /* Alternative 2: ASN1 version */
+  const uint8_t *asn1_ca_file;
+  const uint8_t *asn1_public_cert;
+  const uint8_t *asn1_private_key;
+  int asn1_ca_file_len;
+  int asn1_public_cert_len;
+  int asn1_private_key_len;
+  int asn1_private_key_type; /* E.G. EVP_PKEY_RSA */
```

Comment:
Ideally the key type enum should not be library specific since the rest of the structure is not. Maybe it is a bit tedious to redefine the possible values from evp.h, but I think only a couple of them are really meaningful, so it could be worth it for the sake of consistency.

Reply:
I had added a sample key type as a reminder / hint as the documentation for SSL_CTX_use_PrivateKey_ASN1 is not that helpful.